report/data_sources/hubspot_client.py

```diff
@@ -145,21 +203,16 @@ def _column_mapping_from_csv(cls, csv_file, object_type, primary_key_field):
                 f"Could not find primary key column: {primary_key_field} in CSV file {csv_file}"
             )
 
-
-@dataclass
-class DummyHubspotClient:
-    """A fake Hubspot Client which can be used in tests."""
-
-    companies: List[dict] = None
-
-    def get_companies(self, *args, **kwargs):
-        """Replicates the method from the real object."""
-
-        print("DummyHubspotClient: get_companies() called with", args, kwargs)
-        return self.companies
-
     @classmethod
-    def import_csv(cls, *args, **kwargs):
-        """Replicates the method from the real object."""
-
-        print("DummyHubspotClient: import_csv() called with", args, kwargs)
+    def _get_objects(cls, accessor, fields):
+        for item in accessor.get_all(properties=[field.hs_field for field in fields]):
+            # Filter and map the requested fields
+            result = {}
+            for field in fields:
+                value = item.properties[field.hs_field] or None
+                if value and field.mapping:
+                    value = field.mapping(value)
+
+                result[field.key] = value
```

Comment:
We had some mapping being done inside the object and then different mapping happening outside. I've moved it all inside the client, where it simplifies the calling script and gets tested as well.